File: my_py_pkg/my_py_pkg/robot_news_station.py
import rclpy # ROS library
from rclpy.node import Node
from example_interfaces.msg import String

class RobotNewsStationNode(Node): 
    def __init__(self): 
        super().__init__("robot_news_station")

        # declare parameter for msg.data 
        self.declare_parameter("msg_data", "Hello")

                                                       # robot_news is the name of topic you want to send data to it
        self.publisher = self.create_publisher(String, "robot_news", 10)
        self.timer = self.create_timer(0.5, self.publish_news)
        self.get_logger().info("Robot News Station has been Started")

    def publish_news(self): 
        msg = String()
        msg.data = "Hello"
        self.publisher.publish(msg)

def main(args=None): 
    rclpy.init(args=args) 
    node = RobotNewsStationNode()
    rclpy.spin(node)
    rclpy.shutdown()

if __name__ == "__main__": 
    main()

File: my_cpp_pkg/src/robot_news_station.cpp
#include "rclcpp/rclcpp.hpp" 
#include "example_interfaces/msg/string.hpp"
#include <string>
#include <sstream> 

class RobotNewsStationNode: public rclcpp::Node
{ 
    public:
        RobotNewsStationNode(): Node("robot_news_station")
        {
            RCLCPP_INFO(this->get_logger(), "Hello This is publisher");  

            // decalare robot_name parameter 
            this->declare_parameter<std::string>("robot_name", "sans");

                                                                        //robot_news is the name of topic you want to send data to it
            publisher = this->create_publisher<example_interfaces::msg::String>("robot_news", 10);

            timer_ = this->create_wall_timer(
                std::chrono::seconds(1),
                std::bind(&RobotNewsStationNode::publishNews, this)
            );
        }

    private:
        void publishNews()
        { 
            auto msg = example_interfaces::msg::String();
            std::ostringstream oss;

            robot_name = this->get_parameter("robot_name").as_string();
            oss << "Hi, this is " << robot_name; 

            msg.data = oss.str();
            publisher->publish(msg);  
        }

        std::string robot_name; 
        rclcpp::Publisher<example_interfaces::msg::String>::SharedPtr publisher;
        rclcpp::TimerBase::SharedPtr timer_;
};

int main(int argc, char** argv)
{
    rclcpp::init(argc, argv);
    auto node = std::make_shared<RobotNewsStationNode>(); 
    rclcpp::spin(node);
    rclcpp::shutdown();
    return 0;
}